proposals/jib-cli-surface.md

```diff
@@ -0,0 +1,101 @@
+# Jib CLI Surface Specification
+
+Specification for a command structures exposed to users of the jib-cli
+
+## Usage
+`jib COMMAND [OPTIONS]`
+
+## Commands
+
+The jib cli exposes the following top level commands
+```
+build        build a container
+```
+
+## Options
+
+### Required
+```
+-t  --target <image-ref>  a target image reference, using the url schema defined in jib
+                          gcr.io/project/image (default is build to registry)
+                          registry://gcr.io/project/image
+                          docker://some-image-ref
+                          tar://relative/path.tar
+                          tar:///aboslute/path.tar
+```
+
+### Optional
+
+
+#### Build Config
+```
+    --application-cache <directory>        location of the application cache
+    --base-image-cache <directory>         location of the base image cache
+-b, --build-file <file>                    location of the build file (default <context>/jib.yaml)
+-c, --context <dir>                        location of the build context (default .)
+    --docker-config <directory>            location of docker configuration
+    --name <image-ref>                     image name to bake into tar file (only available when using -t tar://...)
+-p, --parameter <name>=<value>             templating parameters replace `${name}` with `value` in the build file (repeatable)
+    --tag <tag1>[,<tag2>,...]              additional tags for target
+```
+
+#### Auth/Security
+```
+    --allow-insecure-registries            allow jib to communicate with registries over https
+    --send-credentials-over-http           allow jib to send credentials over http (used in conjunction with --allow-insecure-registries)
+```
+##### Credentials
+
+Credentials can be specified using credential helpers or username + password. The following options are available
+```
+    --credHelper <credHelper>              credential helper to use for a registry
```

Comment:
this is all wrong, credential helpers already encode the server url, we do no need to,from,etc for cred helpers

Reply:
Some part of the general Docker auth has bothered me a bit. For example, whether using a credential helper or `docker login`, it didn't seem like you can provide different usernames for the base and the target repos if the repositories are on the same registry. That is, say you have two private Docker Hub repos `alice/image` and `bob/image` (i.e., repos in two different accounts), then you have to repeatedly do `docker login -u alice` and `docker login -u bob` if you need to work on both repos. Using a credential helper doesn't help either; FWIW, credentials are scoped solely by the registry.

Maybe it's just rare in practice that one has two different login accounts on the same registry? Anyways, I think having a single credential helper makes sense. Theoretically feasible, but I think it's unlikely that someone has multiple credential helpers set up only to return different accounts for the same registry.

Reply:
Yeah, I can see that, we can always add in to/from later. It seems strange for now to do it though.